remote/remote_test.go

```diff
@@ -303,97 +419,85 @@ func testImage(t *testing.T, when spec.G, it spec.S) {
 			})
 		})
 	})
 
 	when("#SetEnv", func() {
 		it("sets the environment", func() {
-			img, err := remote.NewImage(repoName, authn.DefaultKeychain)
+			img, err := remote.NewImage(repoName, ggcrauthn.DefaultKeychain)
 			h.AssertNil(t, err)
 
 			err = img.SetEnv("ENV_KEY", "ENV_VAL")
 			h.AssertNil(t, err)
 
 			h.AssertNil(t, img.Save())
 
-			h.AssertNil(t, h.PullImage(dockerClient, repoName))
-			defer h.DockerRmi(dockerClient, repoName)
-
-			inspect, _, err := dockerClient.ImageInspectWithRaw(context.TODO(), repoName)
-			h.AssertNil(t, err)
-
-			h.AssertContains(t, inspect.Config.Env, "ENV_KEY=ENV_VAL")
+			configFile := h.FetchManifestImageConfigFile(t, repoName)
+			h.AssertContains(t, configFile.Config.Env, "ENV_KEY=ENV_VAL")
 		})
 	})
 
 	when("#SetWorkingDir", func() {
 		it("sets the environment", func() {
-			img, err := remote.NewImage(repoName, authn.DefaultKeychain)
+			img, err := remote.NewImage(repoName, ggcrauthn.DefaultKeychain)
 			h.AssertNil(t, err)
 
 			err = img.SetWorkingDir("/some/work/dir")
 			h.AssertNil(t, err)
 
 			h.AssertNil(t, img.Save())
 
-			h.AssertNil(t, h.PullImage(dockerClient, repoName))
-			defer h.DockerRmi(dockerClient, repoName)
-
-			inspect, _, err := dockerClient.ImageInspectWithRaw(context.TODO(), repoName)
-			h.AssertNil(t, err)
-
-			h.AssertEq(t, inspect.Config.WorkingDir, "/some/work/dir")
+			configFile := h.FetchManifestImageConfigFile(t, repoName)
+			h.AssertEq(t, configFile.Config.WorkingDir, "/some/work/dir")
 		})
 	})
 
 	when("#SetEntrypoint", func() {
 		it("sets the entrypoint", func() {
-			img, err := remote.NewImage(repoName, authn.DefaultKeychain)
+			img, err := remote.NewImage(repoName, ggcrauthn.DefaultKeychain)
 			h.AssertNil(t, err)
 
 			err = img.SetEntrypoint("some", "entrypoint")
 			h.AssertNil(t, err)
 
 			h.AssertNil(t, img.Save())
 
-			h.AssertNil(t, h.PullImage(dockerClient, repoName))
-			defer h.DockerRmi(dockerClient, repoName)
-
-			inspect, _, err := dockerClient.ImageInspectWithRaw(context.TODO(), repoName)
-			h.AssertNil(t, err)
-
-			h.AssertEq(t, []string(inspect.Config.Entrypoint), []string{"some", "entrypoint"})
+			configFile := h.FetchManifestImageConfigFile(t, repoName)
+			h.AssertEq(t, configFile.Config.Entrypoint, []string{"some", "entrypoint"})
 		})
 	})
 
 	when("#SetCmd", func() {
 		it("sets the cmd", func() {
-			img, err := remote.NewImage(repoName, authn.DefaultKeychain)
+			img, err := remote.NewImage(repoName, ggcrauthn.DefaultKeychain)
 			h.AssertNil(t, err)
 
 			err = img.SetCmd("some", "cmd")
 			h.AssertNil(t, err)
 
 			h.AssertNil(t, img.Save())
 
-			h.AssertNil(t, h.PullImage(dockerClient, repoName))
-			defer h.DockerRmi(dockerClient, repoName)
-
-			inspect, _, err := dockerClient.ImageInspectWithRaw(context.TODO(), repoName)
-			h.AssertNil(t, err)
-
-			h.AssertEq(t, []string(inspect.Config.Cmd), []string{"some", "cmd"})
+			configFile := h.FetchManifestImageConfigFile(t, repoName)
```

Comment:
Just curious, what is the reason for doing away with `dockerClient.ImageInspectWithRaw`?

Reply:
If we can make all `remote` tests just use direct registry queries (via ggcr) instead of indirectly through `dockerClient.Pull` + `dockerClient.Inspect`, the tests end up being identical for any image OS, run faster, less code and eventually no Daemon dependency.

Reply:
Ah, thank you! That is great!